### PR TITLE
Adding GKE to examples

### PR DESCRIPTION
**Pull request by corey-nm**
Adds link to GKE tutorial on how to deploy a dockerized web server to GKE. I don't think we should duplicate this information in our repo's readme because:
1. The GKE tutorial has a TON of useful commands/help.
2. It makes it explicit that we don't require anything special to run on GKE. This means users with GKE experience already don't need to read through our README just to find out that we are using normal GKE commands.
3. Any docs duplicated here may become out of date.

test plan: built and ran docker image locally

### CODE REVIEW

File: examples/google-kubernetes-engine/server-config.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+num_cores: 1
+num_workers: 1
```
Comment (bfineran) [inline examples/google-kubernetes-engine/server-config.yaml:2]: should remove this as we will use the defaults once added

**Review by bfineran — CHANGES_REQUESTED** [review]
let's add support to the dockerfile to support any server config that comes in. also need to document/support how to mount custom deployment directories to the image that the configs can reference

also should at minimum add a section to the readme demonstrating how to make an inference request to GKE once everything is spun up, even if the rest is through their tutorial

**Comment by corey-nm** [comment]
> let's add support to the dockerfile to support any server config that comes in. also need to document/support how to mount custom deployment directories to the image that the configs can reference
> 
> also should at minimum add a section to the readme demonstrating how to make an inference request to GKE once everything is spun up, even if the rest is through their tutorial

What do you mean by custom deployment directories? I imagine that could be covered in a separate example just going over that with docker.

I added a section to readme, you can interact with it just like any other web server so pretty straightforward!

**Review by bfineran — CHANGES_REQUESTED** [review]
So right now, to use the Dockerfile, users need to edit `server-config.yaml`, or copy the Dockerfile into the directory where their own `server-config.yaml` exists.  Am suggesting that we add support for a config arg along with documentation for it.

Also includes needing to add documentation for how to mount models to the image that the server config can read from local deployment directories. (ie if a user obtains a local deployment directory from sparsify, writes a config for it, then wants to use the dockerfile to deploy to GKE)

File: examples/README.md [hunk 2]
```diff
@@ -32,7 +32,8 @@ Open a Pull Request to [contribute](https://github.com/neuralmagic/deepsparse/bl
 | [Image Classification](https://github.com/neuralmagic/deepsparse/tree/main/examples/classification/)  | How to use image classification models from SparseZoo to perform inference and benchmarking with the DeepSparse Engine  |
 | [Object Detection](https://github.com/neuralmagic/deepsparse/tree/main/examples/detection/)  | How to use object detection models from SparseZoo to perform inference and benchmarking with the DeepSparse Engine  |
 | [Instance Segmentation](https://github.com/neuralmagic/deepsparse/tree/main/examples/dbolya-yolact/)  | How to use an optimized YOLACT model and the DeepSparse Engine to perform real-time instance segmentation. |
-| [AWS Sagemaker Integration](https://github.com/neuralmagic/deepsparse/tree/main/examples/aws-sagemaker/)  | How to deploy a DeepSparse inference server on SageMaker |
+| [AWS Sagemaker Integration](https://github.com/neuralmagic/deepsparse/tree/main/examples/google-kubernetes-engine/)  | How to deploy a DeepSparse inference server on SageMaker |
+| [Google Kubernetes Engine]() | How to deploy a DeepSparse inference server on GKE |
```
[the remark this answers is not in the document]
Comment (corey-nm) [reply examples/README.md:36]: It'll work when it's merged into main

File: examples/google-kubernetes-engine/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,9 @@
+FROM python:3.9
+
+ARG GIT_BRANCH=main
+RUN git clone https://github.com/neuralmagic/deepsparse.git --depth 1 -b $GIT_BRANCH
```
Comment (rahul-tuli) [inline examples/google-kubernetes-engine/Dockerfile:4]: nice!

**Review by rahul-tuli — APPROVED** [review]

File: examples/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (bfineran) [reply examples/README.md:36]: issue isn't branch. it looks like the link for SM is pointing to GKE and the GKE link is missing

[the remark this answers is not in the document]
Comment (corey-nm) [reply examples/README.md:36]: Oooooh woops thank you

[the remark this answers is not in the document]
Comment (corey-nm) [reply examples/README.md]: Fixed

**Review by bfineran — APPROVED** [review]

**Review by KSGulin — APPROVED** [review]

**Pull request merged**